feat(examples): Port basic Linux network utilities (#142)
unikraft/catalog: examples/net-tools/README.md

# Unikraft `net-tools` port

This is a port of `net-tools` for Unikraft as an `examples` category.

> Please note some of the applications ported are not fully functional,
> due to some kernel level functionalities not being yet implemeted by Unikraft.

Reviewer: ```suggestion
Please note some of the applications ported are not fully functional, due to some kernel level functionalities not being yet implemented by Unikraft.
```

Use one sentence per line. Do spell checking.

> This port is mainly used to showcase the capabilities of running network applications on Unikraft
> containers.

Reviewer: ```suggestion
This port is mainly used to showcase the capabilities of running network applications on Unikraft.
It is unlikely there will a real use case for them.
```

Use one sentence per line.

This port contains:

- `curl`
- `dig`
- `host`
- `netcat`
- `mtr`
- `nslookup`
- `ping`
- `traceroute`
- `wget`

In order to use these applications, please modify the `Kraftfile` `cmd` variable to set the program arguments.

# Set Up

To run this example, [install Unikraft's companion command-line toolchain `kraft`](https://unikraft.org/docs/cli), clone this repository and `cd` into this directory.

## Run and Use

Use `kraft` to run the image and start a Unikraft instance. Inside the directory, run:

```bash

> kraft run --rm --plat qemu --arch x86_64 -K net-tools/app/Kraftfile

Reviewer: ```suggestion
kraft run --rm --plat qemu --arch x86_64 -K <app>/Kraftfile
```

```

> Where `app` is the name of the program you want to start.

Reviewer: ```suggestion
where `<app<` is the name of the program you want to start (e.g. `curl`, `dig`, `netcat`).
```
